### DISCUSSION TITLE
Issue with AddHistory Seasons

### DISCUSSION THREAD

**Discussion by michalss** [henrikfroehling/Trakt.NET#426] [Help] [answered]
### Failure description?

```csharp
         var show = await _client.Shows.GetShowAsync(traktID, traktExtended);
                    var seasonA = await _client.Seasons.GetSeasonAsync(traktID, (uint)historyData.extraData.season_number);
                    //var seasons = await _client.Seasons.Get(traktID, (uint)historyData.extraData.season_number);

                    var watchedItem = TraktPost.NewSyncHistoryPost().WithShow(show.Value).WithSeasons(seasonA).Build();
                    await _client.Sync.AddWatchedHistoryItemsAsync(watchedItem);
```

seasonA returning me lTraktEpisode, but it should return ITraktSeason

### Version

1.3.0 (Latest)

### Relevant Stacktrace

well wrong return type

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by henrikfroehling** [comment]
As defined in the [Trakt API Documentation](https://trakt.docs.apiary.io/#reference/seasons/season/get-single-season-for-a-show), a season is a list of episodes.

Therefore
```csharp
TraktListResponse<ITraktEpisode> seasonEpisodesResponse = await _client.Seasons.GetSeasonAsync(traktID, (uint)historyData.extraData.season_number);
```
is correct as it is.

```csharp
var historyPost = TraktPost.NewSyncHistoryPost().WithShow(show.Value).WithSeasons(...).Build();
```
is meant for actual `ITraktSeason` objects as you can get with
```csharp
TraktListResponse<ITraktSeason> seasonsResponse = await _client.Seasons.GetAllSeasonsAsync(traktID);
```
as defined in the [Trakt API Documentation](https://trakt.docs.apiary.io/#reference/seasons/summary/get-all-seasons-for-a-show).

What you are looking for is this:
```csharp
var historyPost = TraktPost.NewSyncHistoryPost().WithShow(show.Value).WithEpisodes(seasonEpisodesResponse).Build();
```
Then, it should work.

**Comment by michalss** [comment]
Problem is i do want to add to history only 1 season. Your example add me all seasons and all episodes. This is nothing i need.. :( Can you please post sample, how to add to history for only single season pls ?

**Comment by michalss** [comment]
Never mind go it.. Thx for help..

**Comment by henrikfroehling** [comment]
You have to get all seasons, without extended info to save data, and pick your specific `ITraktSeason` and add it like this:
```csharp
var historyPost = TraktPost.NewSyncHistoryPost().WithSeason(season).Build();
```

Adding only a season number without any additional info wouldn't work, because then Trakt doesn't know to which show it belongs.
Season alone with IDs (`ITraktSeason` or `ITraktSeasonIDs`) would work.
Season number alone only associated with a show => `WithShowAndSeason()` methods in post builder.
That would look like this:
```csharp
var historyPost = TraktPost.NewSyncHistoryPost()
    .WithShowAndSeason(show,
        new PostHistorySeason
        {
            1 // season number
        })
    .Build();
```

`GetAllSeasons()` might also return less data than `GetSeason()`.

How did you solve your problem?

**Comment by michalss** [comment, marked as answer]
This way :)

```csharp
        var show = await _client.Shows.GetShowAsync(traktID, traktExtended);
                    TraktListResponse<ITraktEpisode> seasonEpisodesResponse = await _client.Seasons.GetSeasonAsync(traktID, (uint)historyData.extraData.season_number);

                    TraktListResponse<ITraktSeason> seasonsResponse = await _client.Seasons.GetAllSeasonsAsync(traktID);
                    var selSeason = seasonsResponse.Where(s => s.Number == (uint)historyData.extraData.season_number).FirstOrDefault();
                    if (selSeason != null)
                    {
                        var historyPost = TraktPost.NewSyncHistoryPost().WithSeason(selSeason).Build();
                        await _client.Sync.AddWatchedHistoryItemsAsync(historyPost);
                    }
```
